File: geolocation/geoloc.py
from flask import Flask, jsonify, request
import csv

app = Flask(__name__)


@app.route("/")
def main():
    return "Welcome to Geolocation!"


@app.route("/get_my_ip", methods=["GET"])
def get_my_ip():
    if request.headers.getlist("X-Forwarded-For"):
        ip = request.headers.getlist("X-Forwarded-For")[0]
    else:
        ip = request.remote_addr
    return jsonify({'ip': ip}), 200


@app.route("/get_my_location", methods=["GET"])
def detect_location():
    if request.headers.getlist("X-Forwarded-For"):
        ip_add = request.headers.getlist("X-Forwarded-For")[0]
    else:
        ip_add = request.remote_addr
    if request.is_json:
        ip_add = request.json["ip"]
    a = [int(x) for x in ip_add.split(".")]
    ip = (a[0] * (256 ** 3)) + (a[1] * (256 ** 2)) + (a[2] * (256 ** 1)) + (a[3] * (256 ** 0))
    with open("IP2LOCATION-LITE-DB1.csv", 'r') as file:
        csvreader = csv.reader(file)
        for row in csvreader:
            if int(row[0]) < int(ip) < int(row[1]):
                start = '.'.join([str(int(row[0]) >> (i << 3) & 0xFF) for i in range(4)[::-1]])
                end = '.'.join([str(int(row[1]) >> (i << 3) & 0xFF) for i in range(4)[::-1]])
                return jsonify(
                    ip_address=ip_add,
                    location=row[2],
                    country=row[3],
                    ip_range_start=start,
                    ip_range_end=end
                )


if __name__ == '__main__':
    app.run(host="localhost", debug=True)

File: geoloc_js/geoloc.js
const express = require('express');
const app = express();
const port = 5000;
const bodyParser = require('body-parser');
app.use(bodyParser.json());

app.get('/', (req, res) => {
  res.send('Welcome to Geolocation!')
})

app.get('/get_my_ip', (req, res) => {
  let ip = req.headers['x-forwarded-for'] || req.socket.remoteAddress
  return res.status(200).json({'Your IP': ip})
})

app.get('/get_my_location', (req, res) => {

  function toIp(int) {
    return ( (int>>>24) +'.'+ (int>>16 & 255) +'.'+ (int>>8 & 255) +'.'+ (int & 255) );
  }

  let ip = req.headers['x-forwarded-for'] || req.socket.remoteAddress;

  let ip1 = ip.split(".");
  let ip2 = (ip1[0] * (256 ** 3)) + (ip1[1] * (256 ** 2)) + (ip1[2] * (256 ** 1)) + (ip1[3] * (256 ** 0));

  let data = require("fs").readFileSync("IP2LOCATION-LITE-DB1.CSV", "utf8");
  data = data.split("\r\n");
  for (let i = 0; i < data.length; i++) {
    data[i] = data[i].replace(/['"]+/g, '').split(",");
  }

  for (let row of data){
    if (+row[0] < ip2 && ip2 < +row[1]) {
      return res.status(200).json({
        "Your IP adress": ip,
        "Location": row[2],
        "Country": row[3],
        "IP range start": toIp(row[0]),
        "IP range end": toIp(row[1])
      })
    }
  }

})

app.listen(port, () => {
  console.log(`App listening at http://localhost:${port}`)
})